homeassistant/components/toon/coordinator.py

```diff
@@ -0,0 +1,132 @@
+"""Provides the Toon DataUpdateCoordinator."""
+import logging
+import secrets
+from typing import Optional
+
+from toonapi import Status, Toon, ToonError
+
+from homeassistant.components.webhook import (
+    async_register as webhook_register,
+    async_unregister as webhook_unregister,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_WEBHOOK_ID, EVENT_HOMEASSISTANT_STOP
+from homeassistant.core import Event, HomeAssistant
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+from homeassistant.helpers.config_entry_oauth2_flow import OAuth2Session
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator, UpdateFailed
+
+from .const import CONF_CLOUDHOOK_URL, DEFAULT_SCAN_INTERVAL, DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class ToonDataUpdateCoordinator(DataUpdateCoordinator):
+    """Class to manage fetching WLED data from single endpoint."""
+
+    def __init__(
+        self, hass: HomeAssistant, *, entry: ConfigEntry, session: OAuth2Session
+    ):
+        """Initialize global Toon data updater."""
+        self.session = session
+        self.entry = entry
+
+        async def async_token_refresh() -> str:
+            await session.async_ensure_token_valid()
+            return session.token["access_token"]
+
+        self.toon = Toon(
+            token=session.token["access_token"],
+            session=async_get_clientsession(hass),
+            token_refresh_method=async_token_refresh,
+        )
```

Comment:
Oh no, you didn't use our guide 😉  https://developers.home-assistant.io/docs/api_lib_auth

Reply:
Right... I found this to be a more elegant from a library perspective personally, but that is a taste I guess.

Is it a problem?

Reply:
Nah, just an observation.